### PR TITLE
Check whether the hash represents a block that exists in the blockcha…

### PR DESCRIPTION
**Pull request by bokobza**
…in before trying to sync.

Fix for https://github.com/stratisproject/FullNodeUI/issues/18

### CODE REVIEW

**Comment by bokobza** [comment]
#FAIL 
Don't merge it in.

**Comment by bokobza** [comment]
Seems to be ok now.

File: src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs [hunk 1]
```diff
@@ -42,28 +42,32 @@ public IActionResult SyncFrom([FromQuery] string from)
...
-
+            
             // Check if an integer was provided as a parameter, meaning the request specifies a block height.
             // If not, the request specified is a block hash.
             bool isHeight = int.TryParse(from, out int height);
             if (isHeight)
             {
-                var chainedBlock = this.chain.GetBlock(height);
-                if (chainedBlock == null)
+                var block = this.chain.GetBlock(height);
```
Comment (noescape00) [inline src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs]: unvar please.

File: src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs [hunk 2]
```diff
@@ -42,28 +42,32 @@ public IActionResult SyncFrom([FromQuery] string from)
...
 
-                hashToSyncFrom = chainedBlock.HashBlock;
+                this.blockNotification.SyncFrom(block.HashBlock);
             }
             else
             {
-                hashToSyncFrom = uint256.Parse(from);
-            }
+                uint256 hashToSyncFrom = uint256.Parse(from);
+                ChainedBlock block = this.chain.GetBlock(hashToSyncFrom);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs]: Ah ofcourse good spot

File: src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs [hunk 3]
```diff
@@ -42,28 +42,32 @@ public IActionResult SyncFrom([FromQuery] string from)
...
                 }
 
-                hashToSyncFrom = chainedBlock.HashBlock;
+                this.blockNotification.SyncFrom(block.HashBlock);
             }
             else
             {
-                hashToSyncFrom = uint256.Parse(from);
-            }
+                uint256 hashToSyncFrom = uint256.Parse(from);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Notifications/Controllers/NotificationsController.cs]: Can `uint256.Parse()`  fail?
Ah well if it fails the controller will throw so thats fine.

**Pull request merged**